pkg/mesos/client_test.go

```diff
@@ -0,0 +1,187 @@
+		assert.NoError(t, json.NewEncoder(w).Encode(expectedHosts))
+	}))
+	defer ts.Close()
+
+	c := NewClient(pluginutil.HTTPClient(ts.URL))
+
+	masters, err := c.Masters()
+	require.NoError(t, err)
+	require.Equal(t, expectedHosts, masters)
+}
+
+func TestState(t *testing.T) {
+	expectedState := State{
+		ID:       "2a2a4995-3a44-41b8-9610-1e0bca965e88",
+		PID:      "10.0.0.10@2a2a4995-3a44-41b8-9610-1e0bca965e88",
+		Hostname: "master.mesos",
+		Cluster:  "cluster-ops-cli-test-cluster-fcdb4f27",
+	}
+
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		assert.Equal(t, "/mesos/master/state", r.URL.String())
+		assert.NoError(t, json.NewEncoder(w).Encode(expectedState))
+	}))
+	defer ts.Close()
+
+	c := NewClient(pluginutil.HTTPClient(ts.URL))
+
+	s, err := c.State()
+	require.NoError(t, err)
+	require.Equal(t, &expectedState, s)
+}
+
+func TestStateSummary(t *testing.T) {
+	expectedState := State{
+		Hostname: "master.mesos",
+		Cluster:  "cluster-ops-cli-test-cluster-fcdb4f27",
+		Slaves: []Slave{
+			Slave{
+				ID:       "2a2a4995-3a44-41b8-9610-1e0bca965e88-S0",
+				PID:      "10.0.0.23@2a2a4995-3a44-41b8-9610-1e0bca965e88-S0",
+				Hostname: "10.0.0.23",
+			},
+		},
+	}
+
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		assert.Equal(t, "/mesos/master/state-summary", r.URL.String())
+		assert.NoError(t, json.NewEncoder(w).Encode(expectedState))
+	}))
+	defer ts.Close()
+
+	c := NewClient(pluginutil.HTTPClient(ts.URL))
+
+	ss, err := c.StateSummary()
+	require.NoError(t, err)
+	require.Equal(t, expectedState.Hostname, ss.Hostname)
+	require.Equal(t, expectedState.Cluster, ss.Cluster)
+	require.Equal(t, expectedState.Slaves, ss.Slaves)
+}
+
+func TestAgents(t *testing.T) {
+	expectedAgents := master.Response{
+		GetAgents: &master.Response_GetAgents{
+			Agents: []master.Response_GetAgents_Agent{
+				master.Response_GetAgents_Agent{
+					Active: true,
+					AgentInfo: mesos.AgentInfo{
+						Hostname: "10.0.0.23",
+					},
+				},
+			},
+		},
+	}
+
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		assert.Equal(t, "/mesos/api/v1", r.URL.String())
+		assert.Equal(t, "application/x-protobuf", r.Header.Get("Accept"))
+		response, err := proto.Marshal(&expectedAgents)
+		assert.NoError(t, err)
+		w.Write(response)
+	}))
+	defer ts.Close()
+
+	c := NewClient(pluginutil.HTTPClient(ts.URL))
+
+	agents, err := c.Agents()
+	require.NoError(t, err)
+	require.Equal(t, expectedAgents.GetAgents.Agents, agents)
+}
+
+func TestMarkAgentGone(t *testing.T) {
+	const expectedAgentID = "9001"
+	expectedBody := master.Call{
+		Type: master.Call_MARK_AGENT_GONE,
+		MarkAgentGone: &master.Call_MarkAgentGone{
+			AgentID: mesos.AgentID{Value: expectedAgentID},
+		},
+	}
+
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		assert.Equal(t, "/mesos/api/v1", r.URL.String())
```

Comment:
could be nice to also assert on the request method being `POST` here, and asserting on `GET` in other tests.